### PR TITLE
Added webpacker and updated rails to 5.1.3

### PR DESCRIPTION
**Pull request by mtejedorwolox**
## Summary

Added webpacker
Updated rails from 5.0.1 to 5.1.3
Updated README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +81,28 @@ Log out from postgres and run:
...
+   rails webpacker:install
+```
+
+Run the Webpacker server with:
+
+```bash
+   ./bin/webpack-dev-server
+```
+
+#### React
```
Comment (mishuagopian) [inline README.md]: I'd change this title to something like `#### Using React In Views`

File: config/webpack/loaders/assets.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const { env, publicPath } = require('../configuration.js')
+
+module.exports = {
```
Comment (mishuagopian) [inline config/webpack/loaders/assets.js]: We had some problems here @alebian. Would you check this for me please?

File: config/webpack/loaders/coffee.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (mishuagopian) [inline config/webpack/loaders/coffee.js]: We don't use coffee anymore, right?

Comment (hdf1986) [reply config/webpack/loaders/coffee.js]: I think we can just remove coffe @mtejedorwolox

File: config/webpack/loaders/assets.js (hunk 2 above)
Comment (hdf1986) [reply config/webpack/loaders/assets.js]: Ping here @alebian

File: Gemfile [hunk 4]
```diff
@@ -77,6 +77,8 @@ gem 'recipient_interceptor'
 # Use for DoS attacks
 gem 'rack-attack'
 
+gem 'webpacker'
```
Comment (hdf1986) [inline Gemfile]: Can we version this? (By the way, i think the version 3.0 of webpacker just went out.
https://twitter.com/dhh/status/902982685945196544

File: README.md [hunk 5]
```diff
@@ -72,6 +81,28 @@ Log out from postgres and run:
...
+Open a terminal to install Webpacker dependencies:
+
+```bash
+   rails webpacker:install
+```
+
+Run the Webpacker server with:
+
+```bash
+   ./bin/webpack-dev-server
```
Comment (hdf1986) [inline README.md]: If we can move to webpacker 3.0, we aren't going to need this :)

File: config/webpack/loaders/coffee.js (hunk 3 above)
Comment (mishuagopian) [reply config/webpack/loaders/coffee.js]: What about this? @mtejedorwolox @hdf1986

Comment (hdf1986) [reply config/webpack/loaders/coffee.js]: Are you spying our conversations?! I am talking this with @mtejedorwolox right now

File: config/webpack/loaders/assets.js (hunk 2 above)
Comment (mtejedorwolox) [reply config/webpack/loaders/assets.js]: Ping @alebian

Comment (alebian) [reply config/webpack/loaders/assets.js]: I have the exact same content

File: package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "resolve-url-loader": "^2.1.0",
+    "sass-loader": "^6.0.6",
+    "style-loader": "^0.18.2",
+    "webpack": "^3.5.5",
+    "webpack-manifest-plugin": "^1.3.1",
+    "webpack-merge": "^4.1.0"
+  },
+  "devDependencies": {
+    "webpack-dev-server": "^2.7.1"
+  }
```
Comment (alebian) [inline package.json]: We added this useful scripts in out config:

```
  "scripts": {
    "dev-server": "./bin/webpack-dev-server --hot --host 127.0.0.1",
    "lint": "eslint ./app/javascript --ext jsx --ext js",
    "lint-fix": "eslint ./app/javascript --ext jsx --ext js --fix"
  }
```

File: package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    "rails-erb-loader": "^5.2.1",
+    "resolve-url-loader": "^2.1.0",
+    "sass-loader": "^6.0.6",
+    "style-loader": "^0.18.2",
+    "webpack": "^3.5.5",
+    "webpack-manifest-plugin": "^1.3.1",
+    "webpack-merge": "^4.1.0"
+  },
+  "devDependencies": {
+    "webpack-dev-server": "^2.7.1"
```
Comment (alebian) [inline package.json]: We should add eslint, right @sbalay 

This is our conf
```
    "babel-eslint": "^7.2.3",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "eslint": "^3.19.0",
    "eslint-config-airbnb": "^15.0.1",
    "eslint-config-prettier": "^2.1.1",
    "eslint-plugin-flowtype": "^2.34.0",
    "eslint-plugin-import": "^2.3.0",
    "eslint-plugin-jsx-a11y": "^5.0.3",
    "eslint-plugin-prettier": "^2.1.1",
    "eslint-plugin-react": "^7.0.1",
    "prettier": "^1.4.0",
    "prettier-eslint": "^6.2.3",
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
+{
+  "dependencies": {
```
Comment (alebian) [inline package.json]: What do you guys think of adding basic libraries as dependencies like: axios, bootstrap, immutable, jquery, etc @mishuagopian @sbalay

File: package.json (hunk 6 above)
Comment (hdf1986) [reply package.json]: We aren't going to need the dev-server script, webpack 3.0 includes the dev-server inside the standard rails server.

Comment (hdf1986) [reply package.json]: @mtejedorwolox @leamotta Can we include the rest of the scripts?

File: package.json (hunk 7 above)
Comment (sbalay) [reply package.json]: Yeah, @mtejedorwolox use the rules from here https://github.com/Wolox/wolmo-bootstrap-react-native/blob/master/generators/app/templates/.eslintrc.ejs

Comment (sbalay) [reply package.json]: @mtejedorwolox you will also need to add the babel plugins to the babel config.

**Comment by francoprud** [comment]
I think we must add this linter commented in the pre-push script.

**Comment by mtejedorwolox** [comment]
@francoprud @hdf1986 please check the new react script

File: script/react [hunk 9]
```diff
@@ -5,7 +5,7 @@ if [[ "$unamestr" == 'Darwin' ]]; then
   #MacOS
   sed -i ".original" "/gem 'react-rails'/d" "Gemfile"
   sed -i ".original" '$!N;/^\n$/!P;D' "Gemfile"
-  sed -i ".original" "s/gem 'webpacker'\(, '~> [0-9]\(\.[0-9]\(\.[0-9]\)\?\)\?'\)\?/&\n\ngem 'react-rails'/g" "Gemfile"
+  sed -i ".original" $'s/gem \'webpacker\'.*$/&\\\n\\\ngem \'react-rails\'/g' "Gemfile"
   bundle install
   rails webpacker:install:react
   sed -i ".original" "s/body$/  = javascript_pack_tag 'application'\n  &/g" "app/views/layouts/application.html.slim"
```
Comment (francoprud) [inline script/react]: @mtejedorwolox check this line. It has the same problem as before. Only one step more and 👍

**Comment by mishuagopian** [comment]
@hdf1986 @francoprud @mtejedorwolox ping here! Is this ready?

**Comment by hdf1986** [comment]
@francoprud i've just added the linter to the prepush script. Also i updated the react-rails reference in the React script, the version that's on rubygems breaks with webpacker 3.0, so'll need to use the github version for now. @mishuagopian This is ready, take a look when you have some free time

File: .gitignore [hunk 10]
```diff
@@ -43,3 +43,8 @@ dump.rdb
 
 # Ignore .env files
 .env*
+/public/packs
+/node_modules
+/public/packs
+/public/packs-test
+/node_modules
```
Comment (mishuagopian) [inline .gitignore]: Remove repetion

File: .gitignore [hunk 11]
```diff
@@ -43,3 +43,8 @@ dump.rdb
 
 # Ignore .env files
 .env*
+/public/packs
+/node_modules
+/public/packs
```
Comment (mishuagopian) [inline .gitignore]: Remove repetion

**Pull request merged**